extension.js

```diff
@@ -356,10 +368,31 @@ function update_height_setting(win) {
     settings.set_double('window-height', Math.min(1.0, current_height));
 }
 
+function set_window_maximize() {
+    if (!current_window)
+        return;
+
+    const should_maximize = settings.get_boolean('window-maximize');
+    if (current_window.maximized_vertically && !should_maximize)
+        current_window.unmaximize(Meta.MaximizeFlags.VERTICAL);
+    else if (!current_window.maximized_vertically && should_maximize)
+        current_window.maximize(Meta.MaximizeFlags.VERTICAL);
+}
+
+function update_window_height() {
+    if (settings.get_boolean('window-maximize')) {
+        settings.set_boolean('window-maximize', false);
+        settings.set_double('window-height', 1.0);
```

Comment:
Is setting `window-height` really necessary here?

BTW, it will trigger another `update_window_height()` if `window-height` isn't `1.0` already (not sure if later from event loop, or recursively).